omegaconf/_utils.py

```diff
@@ -195,13 +196,15 @@ def _resolve_forward(type_: Type[Any], module: str) -> Type[Any]:
 def get_attr_data(obj: Any, allow_objects: Optional[bool] = None) -> Dict[str, Any]:
     from omegaconf.omegaconf import OmegaConf, _maybe_wrap
 
+    obj_type = get_type_of(obj)
+
     flags = {"allow_objects": allow_objects} if allow_objects is not None else {}
     dummy_parent = OmegaConf.create(flags=flags)
+    dummy_parent._metadata.object_type = obj_type
```

Comment:
Is this change still required?

Reply:
Nope. Reverted in https://github.com/omry/omegaconf/pull/559/commits/703b6df88795df9b93e1dfc4c76bc135a594e6a0.